### ISSUE TITLE
scroll into view seems to return too early

### ISSUE THREAD

**Issue by svettwer** [sakuli/sakuli#411]
I recognized that some `_click` action only succeed, if a `_highlight` is used before. It seems that the scroll into view function returns before the scrolling has been finished. Therefore the test tries to click an element that is still moving and therefore fails.

**Acceptance criteria**
- [ ] The click is executed successfully even if no `_highlight` is used

**Additional information**
```javascript
(async () => {
  const testCase = new TestCase("");
  try {
    await _navigateTo("http://oldtimer.de");
    await _wait(10000, () => _isVisible(_div("WW_ct")));
    if (await _isVisible(_button(/.*WW_ui-cssbtn-close.*/)) === true) {
      await _click(_button(/.*WW_ui-cssbtn-close.*/));
    }
    await _click(_link(/.*Beitrag berechnen.*/));
    await _wait(10000, () => _isVisible(_iframe("OldtimerIfr")));
    await driver.switchTo().frame(0);
    await _wait(10000, () => _isVisible(_div("radioTable")));

    //################
    // await _highlight(_radio("AnnahmeJaNeintrue")); //comment in to fix test
    await _click(_radio("AnnahmeJaNeintrue"));
    //################
    await _wait(2000);

  } catch (e) {
    await testCase.handleException(e);
  } finally {
    await testCase.saveResult();
  }
})();

```

**Comment by tnobody** [comment]
You can affect the behavior of `scrollIntoView` with the CSS property `scroll-behavior`.

The page [http://oldtimer.de] sets this property on the `<HTML>` tag which causes all auto-scroll interaction on the page (triggered by JS `scrollIntoView` or clicking an anchor-link) to scroll smoothly.

You can override this CSS Rule dynamically. I tested it with a quick and dirty hack solution within the actual test case:

```typescript
const resetScrollBehavior = async () => {
  const style = `
    * {
        scroll-behavior: unset !important;
    }
`;
  await _eval(`
    var styleTag = document.createElement('style')
    styleTag.appendChild(
        document.createTextNode(\`${style}\`)
    )
    styleTag.type = 'text/css';
    document.head.appendChild(styleTag)
`);
};

(async () => {
  const testCase = new TestCase("");
  try {
    await _navigateTo("http://oldtimer.de");

    await resetScrollBehavior()

    await _wait(10000, () => _isVisible(_div("WW_ct")));
    if ((await _isVisible(_button(/.*WW_ui-cssbtn-close.*/))) === true) {
      await _click(_button(/.*WW_ui-cssbtn-close.*/));
    }
    await _click(_link(/.*Beitrag berechnen.*/));
    await _wait(10000, () => _isVisible(_iframe("OldtimerIfr")));

    await resetScrollBehavior()

    await driver.switchTo().frame(0);
    await _wait(10000, () => _isVisible(_div("radioTable")));

    //################
    // await _highlight(_radio("AnnahmeJaNeintrue")); //comment in to fix test
    await _click(_radio("AnnahmeJaNeintrue"));
    //################
    await _wait(2000);
  } catch (e) {
    await testCase.handleException(e);
  } finally {
    await testCase.saveResult();
  }
})();

```

**Comment by tnobody** [comment]
An alternative approach would be to somehow detect the scroll ending as described in this post: https://gomakethings.com/detecting-when-a-visitor-has-stopped-scrolling-with-vanilla-javascript/

This would not break the page behavior at all but might be little unstable due to the timeout threshold we have to find (in the article the author uses 66ms but doesn't explain why 🤷‍♂ )

**Comment by svettwer** [comment]
I would go with the js script attempt as it is independent from the DOM. My concern altering the website code to add styles for scrolling is, that someone might have added some `!important` statements as well to enforce for e.g. `smooth` scrolling on the element we want to scroll to. I mean there might be no obvious reason from our perspective to do so but the web is a wild place. 
Therefore I'd go with the vanilla javascript attempt as it would not be affected by some "special" styles.

**Comment by svettwer** [comment]
During planning poker, we agreed to create an POC issue for the javascript based attempt.

**Comment by svettwer** [comment]
closes with #421
